src/Expensify.js

```diff
@@ -78,10 +77,14 @@ class Expensify extends PureComponent {
     }
 
     componentDidMount() {
-        Onyx.connect({
-            key: ONYXKEYS.SESSION,
-            callback: this.removeLoadingState,
-        });
+        // Run any Onyx schema migrations and then connect to Onyx
+        MigrateOnyx()
+            .then(() => {
+                Onyx.connect({
+                    key: ONYXKEYS.SESSION,
+                    callback: this.removeLoadingState,
+                });
+            });
```

Comment:
That's good to know. What will trigger the fetching of the session in that case then?

Reply:
It will just be connected `withOnyx()` there is no "loading" we'll either have an `authToken` or we won't

Reply:
OK, just thinking through this with you, but there is only one option to "move it higher up the chain" that I can think of:

```
Migrate().then(() => Onyx.init());
```

The problem here is that `Onyx.init()` is outside of the React lifecycle, and that's OK because it's a synchronous operation. However, since `Migrate()` is asynchronous, that means that this all needs to be moved into the React lifecycle, something like this:

```
class Expensify  {
    constructor() {
        this.state = {onyxInitialized: false};
    }
    componentDidMount() {
        Migrate()
            .then(() => {
                Onyx.init();
                this.setState({onyxInitialized, true});
            })
    }
    render() {
        if (!this.state.onyxInitialized) return null;
    }
}
```

This would be the only way to ensure the app can load after all migrations have run.

Reply:
That looks correct to me.

Reply:
> Is this only because we are using Onyx to perform the migrations?

Yes, exactly. My only thought about why it would be nice to have Onyx initialized before the migration is because since the migrations use Onyx to migrate the data, it seems logical to me that Onyx should be in its initialized state (but there is no hard requirement for this).

I think we can both agree that the stuff happening in `init()` today will have no implication on the current migration that exists, so let's just do nothing in regards to `init()` for now. Otherwise, we're just making guesses based on nothing.

Reply:
That's fair. I'm good to resolve this for now as I can't think of any reason to do this and might just be experiencing some kind of irrational correctness bias.